autoload/neomake/makers/ft/php.vim

```diff
@@ -41,8 +41,16 @@ function! neomake#makers#ft#php#phpcs() abort
 endfunction
 
 function! neomake#makers#ft#php#phpmd() abort
+    let l:args = ['%:p', 'text']
+
+    if exists('g:neomake_php_phpmd_args_rules')
+        call add(l:args, g:neomake_php_phpmd_args_rules)
+    else
+        call add(l:args, 'codesize,design,unusedcode,naming')
+    endif
+
     return {
-        \ 'args': ['%:p', 'text', 'codesize,design,unusedcode,naming'],
+        \ 'args': args,
         \ 'errorformat': '%E%f:%l%\s%m'
         \ }
 endfunction
```

Comment:
Dunno?!
If you mean that having an extra var makes it less likely to cause user errors, then that  might be a point.
However, adding per-maker config settings is not going to scale really.
I've thought about something like `g:neomaker_FT_MAKER_extra_args` that would be appended (since people tend to wrongly overwrite `_args` or just `_maker` itself accidentally, and getting the defaults is long (`neomake#makers#ft#php#phpmd().args`).

`tox` has an interesting concept of a `posargs` setting, which can have a default.
In this case it would default to default list, and could be changed by a user.
OTOH with other makers, you would want to configure `--exclude`, `--include` etc instead (and not just a single arg).